src/Parse.cpp

```diff
@@ -252,28 +280,33 @@ void Parse::ParseAV(char **av){
 		}
 		else if (args.front() == "--heuristic"){
 			args.pop();
+			if (args.empty())
+				throw CustomError("Error: No given argument for --heuristic");
 			if (args.front() == "manhattan")
 				this->_heuristics = MANHATTAN;
-			else if (args.front() == "other")
-				this->_heuristics = OTHER;
-			else if (args.front() == "othertwo")
-				this->_heuristics = OTHERTWO;
+			else if (args.front() == "linear")
+				this->_heuristics = LINEAR_CONFLICT;
```

Comment:
You should make _heuristics a ptr to heuristic type and put directly the function here